pkg/chart/dependency.go

```diff
@@ -22,20 +22,28 @@ type dependencies struct {
 }
 
 // syncDependencies takes care of updating dependencies to correct version and sync to target repo if necesary.
-func syncDependencies(chartPath string, sourceRepo *api.Repo, target *api.TargetRepo, sourceIndex *helmRepo.IndexFile, targetIndex *helmRepo.IndexFile, syncDeps bool) error {
+func syncDependencies(chartPath string, sourceRepo *api.Repo, target *api.TargetRepo, sourceIndex *helmRepo.IndexFile, targetIndex *helmRepo.IndexFile, apiVersion string, syncDeps bool) error {
 	klog.V(3).Info("Chart has dependencies...")
 	var errs error
 	var missingDependencies = false
-	requirementsLockFile := path.Join(chartPath, "requirements.lock")
+	lockFileName := ""
+	if apiVersion == "v1" {
+		lockFileName = "requirements.lock"
+	} else if apiVersion == "v2" {
+		lockFileName = "Chart.lock"
+	} else {
+		return errors.Errorf("unrecognised apiVersion %s", apiVersion)
+	}
+	lockFile := path.Join(chartPath, lockFileName)
```

Comment:
I would add a small function for this (and a test separately)

```
func lockFilePath(chartPath, apiVersion) (string, error) {
}
```

```
err, lockFile := lockFilePath(chartPath, apiVersion)
if err != nil { ... }
```

Reply:
done!